src/bindings/jna/libelektra4j/src/main/java/org/libelektra/exception/mapper/ExceptionMapperService.java

```diff
@@ -0,0 +1,49 @@
+package org.libelektra.exception.mapper;
+
+import org.libelektra.Key;
+import org.libelektra.exception.*;
+
+public class ExceptionMapperService {
+
+	public static KDBException getMappedException(Key k) {
+
+		// At least one error or warning should be there
+		if (k.getMeta("error/number").isNull() && k.getMeta("warnings/#00/number").isNull()) {
+			Key temporaryError = Key.create("user/temporary/errorkey");
+			temporaryError.setMeta("error/number", InternalException.errorCode());
+			temporaryError.setMeta("error/reason", "Sorry, Elektra returned an error/warning but no " +
```

Comment:
Undocumented hack to rewrite the reason? Why not append the keys when returning the message?

Reply:
> Undocumented hack to rewrite the reason?

It is not to "rewrite" the reason but to actually give one because Elektra did not provide one. E.g., you make a call to Elektra, it returns -1 but the errorkey is empty. This should never happen in reality but I provided a fallback so the user would not receive a NullPointer/ Empty Message

Reply:
If you wish I can add another constructor which expects (String errorCode, String reason) as paramters

Reply:
I do not think that another constructor solves the problem.

> you make a call to Elektra, it returns -1 but the errorkey is empty.

Then this is the wrong place to fix this. This code must be within the `kdb*` functions.

Reply:
This is not "fix" per se but a safeguard. There may always be changes in the kdb functions which may cause no error to be returned. The java binding would at least catch this error and not return a null pointer. I already had this behavior when trying to load a plugin which was not existent on the build server.

Reply:
> I do not think that another constructor solves the problem.

The additional constructor would be just some better usability like a constructor with a predefined size I have done in the KeySet class

Reply:
https://de.wikipedia.org/wiki/Design_by_contract

Clients should be able to rely on that they get an error message if they get `-1`. If this is not the case, we have a bug, please report it.

Reply:
I know that it is design by contract. If the contract is violated though it is still good working code. I will simply write "please report a bug to libelektra..." instead.

Reply:
> I know that it is design by contract. If the contract is violated though it is still good working code.

Then please also describe your interface according design-by-contract. I did not see that you try to avoid to return null pointers.

> I will simply write "please report a bug to libelektra..." instead.

Yes, better.

Reply:
> Yes, better.

Actually I see that the current error message is already in this form (did not see it in the marked PR line immediately) `Sorry, Elektra returned an error/warning but no information along it, please report this incident at https://issues.libelektra.org/`